### DISCUSSION TITLE
[Analytical Table]: Show subRow data fetched async on first click

### DISCUSSION THREAD

**Discussion by kmanisha29** [UI5/webcomponents-react#5622] [Q&A]
For a tree table when we provide data and have to fetch subRows and add to data on expand click, how to get get expand button if subrows not provided.
Currently I have to append child IDs in array to subRows of data to get expand icon and on expand click I fetch data from API and update.
Another issue: First click on expand does not open and show proper data when API is called. From next time since data is already present in tree, it expands and collapses properly. Why do we need  2 clicks. Some configuration missing?

**Comment by Lukas742** [comment]
Hi @kmanisha29 

> For a tree table when we provide data and have to fetch subRows and add to data on expand click, how to get get expand button if subrows not provided.
Currently I have to append child IDs in array to subRows of data to get expand icon and on expand click I fetch data from API and update.

this behavior is not supported out of the box, but can most probably be achieved by creating a [custom plugin hook](https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-analyticaltable-recipes--docs#how-to-add-your-own-plugin-hooks). Within the hook, you can set the `canExpand` property of the respective row to `true` which will display the expand button. I've created a quick and very basic example which you can find [here](https://stackblitz.com/edit/github-pm9c3g-vrwmmp?file=src%2Fdata.json,src%2FApp.tsx).

> Another issue: First click on expand does not open and show proper data when API is called. From next time since data is already present in tree, it expands and collapses properly. Why do we need 2 clicks. Some configuration missing?

This sounds like its the same issue as described [here](https://github.com/SAP/ui5-webcomponents-react/issues/5515), but without an isolated example it's unfortunately not possible for us to assist you with it. It would be very helpful if you could provide one so we can start our investigation.

**Reply by kmanisha29** [reply]
Hi @Lukas742 ,

https://stackblitz.com/edit/github-3wiuce?file=src%2FApp.tsx,package.json,src%2FuseData.ts
 Here is a short example created with dummy data only.   Even here resetting data to have subRows does not expand rows first time

**Reply by Lukas742** [reply]
Since you're changing the `data` the internal expanded state is reset. To prevent this you can set `autoResetExpanded` inside the `reactTableOptions` to `false`. You can find out more about this [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-analyticaltable-recipes--docs#how-to-stop-the-table-state-from-automatically-resetting-when-the-data-changes).

https://stackblitz.com/edit/github-3wiuce-hsseaj?file=src%2FApp.tsx
